### ISSUE TITLE
The list items displaying with incorrect order under the list tree node in Inspect when expanding the ImageAlign/TextAlign/Dock properties comboBox in PropertyGrid 

### ISSUE THREAD

**Issue by Zheng-Li01** [dotnet/winforms#5234]
* .NET Core Version: 
.NET 6.0.100-preview.7.21363.2
* Have you experienced this same bug with .NET Framework?: 
 No, the feature is no supported in the .NET Framework & .NET Core 3.1, see details in the GH issue [3027](https://github.com/dotnet/winforms/issues/3027)

**Problem description:**
The list items displaying with incorrect order under the list tree node in Inspect when expanding the ImageAlign/TextAlign/Dock properties comboBox in PropertyGrid. 

**Expected behavior:**
The list items displaying with correct order under the list tree node in Inspect when expanding the ImageAlign/TextAlign/Dock properties comboBox in PropertyGrid. 

**Actual behavior:**
The list items displaying with incorrect order under the list tree node in Inspect when expanding the ImageAlign/TextAlign/Dock properties comboBox in PropertyGrid. 
![image](https://user-images.githubusercontent.com/38325459/125548876-642bc518-7f6c-4283-b6cc-54ad7e209a17.png)

**Minimal repro:**
1. Open the attached application.
[WinFormsApp_Issue1.zip](https://github.com/dotnet/winforms/files/6812986/WinFormsApp_Issue1.zip)
2. Build and run it.
3. Expanding the ImageAlign/TextAlign/Dock properties comboBox in PropertyGrid. 
4. Open the Inspect tool, and observe the list items displaying order under the list tree node in Inspect.

**More info:** 
1. The issue also can reproduce .NET Core 5.0, can't reproduce .NET Core 3.1.
2. The issue can't reproduce others properties, such as BackColor/ForeColor.
3. The issue also can reproduce AccessibilityInsigt tool.

**Comment by merriemcgaw** [comment]
This would be considered a MAS violation. We should get it fixed. @Zheng-Li01 do we have this problem with the Framework version of the PropertyGrid as well, out of curiosity?

**Comment by Zheng-Li01** [comment]
As mentioned above that the issue can't reproduce with the .NET Framework version of the propertyGrid due to there are no list items displaying under the list tree node in Inspect .
![image](https://user-images.githubusercontent.com/38325459/126250207-0c3a4e9c-bfbf-4885-a23e-320be7b838f9.png)

**Comment by NikitaSemenovAkvelon** [comment]
Order in Inspect depends on the order in that controls were added to a Form/Control Collection. Suddenly, they were added in the revert direction (e.g. control3, control 2, control 1) and it is a general practice. @vladimir-krestov, @RussKie do you know for what it was done?
So, I can correct the order of adding controls to fix the order in Inspect, but I don't how it may affect other behavior.

**Comment by merriemcgaw** [comment]
I don't think we need to fix this if it's just incorrect in Inspect. That particular tool is basically deprecated. I wouldn't risk changing anything that might impact any other behavior here.
